ui-core/src/stories/Select.stories.tsx

```diff
@@ -1,20 +1,27 @@
 import React from 'react';
 import { StoryObj, Meta } from '@storybook/react';
 
-import Select from '../components/Select';
+import Select, { SelectProps } from '../components/Select';
 import '@osrd-project/ui-core/dist/theme.css';
 
+type Option = { value: string; label: string };
+
+const options = [
+  { value: 'blue', label: 'Blue' },
+  { value: 'red', label: 'Red' },
+  { value: 'green', label: 'Green' },
+] as Option[];
+
 const meta: Meta<typeof Select> = {
   component: Select,
   args: {
     label: 'Fill colour',
     placeholder: 'Choose',
-    value: 'blue',
-    options: [
-      { value: 'blue', label: 'Blue' },
-      { value: 'red', label: 'Red' },
-      { value: 'green', label: 'Green' },
-    ],
+    value: options[0],
+    options,
+    getOptionLabel: (option: unknown) => (option as Option).label,
+    getOptionValue: (option: unknown) => (option as Option).value,
+    onChange: (option?: unknown) => console.log(option),
```

Comment:
I don't really like this, can you rather define a wrapper like this ? 
```
const SelectWrapper = (props: SelectProps<Option>) => {
  return <Select {...props} />;
};
```
You can use it instead of the Select l16 and then change your functions:
```suggestion
    getOptionLabel: (option: Option) => option.label,
    getOptionValue: (option: Option) => option.value,
    onChange: (option?: Option) => console.log(option),
```